kof-operator/internal/k8s/prometheus.go

```diff
@@ -0,0 +1,42 @@
+package k8s
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/k0rdent/kof/kof-operator/internal/models/target"
+	v1 "github.com/prometheus/prometheus/web/api/v1"
+)
+
+var PrometheusPort string
+
+const PrometheusEndpoint = "api/v1/targets"
+
+func CollectPrometheusTargets(ctx context.Context, logger *logr.Logger, kubeClient *KubeClient, clusterName string) (*target.PrometheusTargets, error) {
+	response := &target.PrometheusTargets{Clusters: make([]*target.Cluster, 0)}
+
+	podList, err := GetCollectorPods(ctx, kubeClient.Client)
+	if err != nil {
+		return response, fmt.Errorf("failed to list pods: %v", err)
+	}
+
+	for _, pod := range podList.Items {
+		byteResponse, err := Proxy(ctx, kubeClient.Clientset, pod, PrometheusPort, PrometheusEndpoint)
+		if err != nil {
+			logger.Error(err, "failed to connect to the pod", "podName", pod.Name)
```

Comment:
Missing valuable information here:
```suggestion
			logger.Error(err, "failed to connect to the pod", "podName", pod.Name, "response", string(byteResponse))
```